src/index.js

```diff
@@ -883,34 +929,49 @@ export default function (babel) {
       },
 
       ArrayComprehension(path) {
-        // disallow Yield and Return
-        path.get("loop.body").traverse({
-          AwaitExpression(awaitPath) {
-            throw awaitPath.buildCodeFrameError(
-              "`await` is not allowed within Comprehensions; " +
-              "instead, await the Comprehension (eg; `y <- [for x of xs: x]`)."
-            );
-          },
-          YieldExpression(yieldPath) {
-            throw yieldPath.buildCodeFrameError("`yield` is not allowed within Comprehensions.");
-          },
-          ReturnStatement(returnPath) {
-            throw returnPath.buildCodeFrameError("`return` is not allowed within Comprehensions.");
-          },
+        validateComprehensionLoopBody(path.get("loop.body"));
```

Comment:
Moving to exit visitor caused some pretty horrific regressions. Stuff like this:
```
Error: /Users/wcj/dev/node_modules/babel-plugin-lightscript/test/fixtures/comprehensions/closure-nested-semantic.js: We don't know what to do with this node type. We were previously a Statement but we can't fit in here?
      at NodePath.insertBefore (node_modules/babel-traverse/lib/path/modification.js:61:13)
```